pkg/template/client/internalversion/templateconfig.go

```diff
@@ -0,0 +1,32 @@
+package internalversion
+
+import (
+	templateapi "github.com/openshift/origin/pkg/template/apis/template"
+	templateinternalversion "github.com/openshift/origin/pkg/template/generated/internalclientset/typed/template/internalversion"
+)
+
+// TemplateConfigInterface is an interface for processing template client.
+type TemplateProcessorInterface interface {
+	Process(*templateapi.Template) (*templateapi.Template, error)
+}
+
+// NewTemplateProcessorClient returns a client capable of processing the templates.
+func NewTemplateProcessorClient(c templateinternalversion.TemplateInterface, ns string) TemplateProcessorInterface {
+	return &templateProcessor{client: c, ns: ns}
+}
+
+type templateProcessor struct {
+	client templateinternalversion.TemplateInterface
```

Comment:
Looking at the package dependencies, I think this is going to have to be a rest.Interface

Reply:
@deads2k sure, we can pass RESTClient() to the initializer, sounds good.

Reply:
> @deads2k sure, we can pass RESTClient() to the initializer, sounds good.

still outstanding.